src/server/db/repository/assessmentCycle/country/getCountryLastPublishedInfo.ts

```diff
@@ -1,22 +1,42 @@
+import { Dates } from 'utils/dates'
+
 import { LastPublishedInfo } from 'meta/area/country'
 import { CountryIso } from 'meta/area/countryIso'
 import { CountryStatus } from 'meta/area/countryStatus'
 import { Assessment } from 'meta/assessment/assessment'
+import { Cycle } from 'meta/assessment/cycle'
+import { Cycles } from 'meta/assessment/cycles'
 
 import { BaseProtocol, DB } from 'server/db/db'
 import { Schemas } from 'server/db/schemas'
 
-type Props = { assessment: Assessment; countryIso?: CountryIso }
+type Props = {
+  assessment: Assessment
+  countryIso?: CountryIso
+  cycle: Cycle
+}
 
 type LastPublishedRecord = Record<CountryIso, LastPublishedInfo>
 
 export const getCountryLastPublishedInfo = async (
   props: Props,
   client: BaseProtocol = DB
 ): Promise<LastPublishedRecord> => {
-  const { assessment, countryIso } = props
+  const { assessment, countryIso, cycle } = props
+
+  const cycleDateCreated = Dates.parseISO(cycle.props.dateCreated)
+
+  let include = true
```

Comment:
can we return true/false from fn without defining it outside?

Reply:
That's iterative . A published cycle can have successor cycles not published . They all must be excluded